src/w3d/lib/sharebuf.h

```diff
@@ -66,3 +69,9 @@ template<class T> class ShareBufferClass : public W3DMPO, public RefCountClass
     T *m_array;
     int m_count;
 };
+
+#ifndef BUILD_EDITOR
+#define NEW_SHARE_BUFFER(T, V) (new ShareBufferClass<T>(V))
+#else
+#define NEW_SHARE_BUFFER(T, V, N) (new ShareBufferClass<T>(V, N))
+#endif
```

Comment:
I suggest to create small helper function instead.

```c++
template <typename T>
ShareBufferClass<T>* New_Share_Buffer_Class(int count, const char* name)
{
#ifndef BUILD_EDITOR
    return new ShareBufferClass<T>(count);
#else
    return new ShareBufferClass<T>(count, name);
#endif
}
```

Same approach for the other new macros. This will be inlined and `name` will be omitted in Release because not used.

Reply:
Further I suggest to use `#ifdef` instead of `#ifndef`. It is easier to the eyes to use non negated conditions. Suggested code updated accordingly.